hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECBlockOutputStreamEntry.java

```diff
@@ -338,30 +334,16 @@ private boolean isFailed(
         = null;
     try {
       containerCommandResponseProto = chunkWriteResponseFuture != null ?
-          chunkWriteResponseFuture.get() :
-          null;
+          chunkWriteResponseFuture.get() : null;
     } catch (InterruptedException e) {
       outputStream.setIoException(e);
       Thread.currentThread().interrupt();
     } catch (ExecutionException e) {
       outputStream.setIoException(e);
     }
 
-    if ((outputStream != null && containerCommandResponseProto != null)
-        && (outputStream.getIoException() != null || isStreamFailed(
-        containerCommandResponseProto, outputStream))) {
-      return true;
-    }
-    return false;
-  }
-
-  boolean isStreamFailed(
-      ContainerProtos.ContainerCommandResponseProto responseProto,
-      ECBlockOutputStream stream) {
-    try {
-      ContainerProtocolCalls.validateContainerResponse(responseProto);
-    } catch (StorageContainerException sce) {
-      stream.setIoException(sce);
+    if (containerCommandResponseProto != null
```

Comment:
I think there are cases when this response can be null ( I am not 100% sure, what case it will be null). In the case of null also, should we consider stream as failed?
@cchenax faced an issue and in his debug logs he was claiming one of future response was null. SO, we may need to rethink here, what if response future null?

And let's add some debug info in this response handling path?

Reply:
Oh, I'm curious to hear this problem and wonders why the future could be null, but the current path doesn't handle this 'future == null' case indeed.
I think if it really happens, we could show debug info here as you suggested and consider the stream as failed.
And let's help @cchenax dig into the issue he faced to get the root cause.

Reply:
While helping @cchenax to debug on the issue for root cause, should we go ahead committing this by adding more debug logs? If you agree, please add few debug logs as we are touching these files.

Reply:
Sure, a simple check for null and debug log is inserted above.

Reply:
Oh, I'm fine with stricter checks for null here, it seems there's a complex problem that chen has hit.
Will update soon.